core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/StackV4Endpoint.java

```diff
@@ -269,4 +273,17 @@ FlowIdentifier setClusterMaintenanceMode(@PathParam("workspaceId") Long workspac
     @ApiOperation(value = UPDATE_SALT, nickname = "updateSaltByName")
     FlowIdentifier updateSaltByName(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name);
 
+    @POST
+    @Path("{name}/database_backup")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = DATABASE_BACKUP, nickname = "databaseBackup")
+    BackupV4Response backupDatabaseByName(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name,
+            @QueryParam("backupLocation") String backupLocation, @QueryParam("backupId") String backupId);
+
+    @POST
+    @Path("{name}/database_restore")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = DATABASE_RESTORE, nickname = "databaseRestore")
+    RestoreV4Response restoreDatabaseByName(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name,
+            @QueryParam("backupLocation") String backupLocation, @QueryParam("backupId") String backupId);
```

Comment:
the same here.. is it uuid or name?

Reply:
backupId is the UUID for the backup. It is the Id generated by datalake-dr service. Yes, the name is the stack name. There is no use for backup name here.